### PR TITLE
Added a reroute call when updating the maxLatency flow field.

### PR DESCRIPTION
**Pull request by dpoltavets**

### CODE REVIEW

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java [hunk 1]
```diff
@@ -279,9 +288,43 @@ public UnidirectionalFlow updateFlow(FlowDto flow) throws FlowNotFoundException
...
             flowRepository.createOrUpdate(currentFlow);
 
-            return Optional.of(forwardFlow);
+            return Optional.of(result.updatedFlow(forwardFlow).build());
 
         }).orElseThrow(() -> new FlowNotFoundException(flow.getFlowId()));
+
+        if (updateFlowResult.isNeedRerouteFlow()) {
+            Flow updatedFlow = updateFlowResult.getUpdatedFlow().getFlow();
+            Set<PathId> pathIds = new HashSet<>(updatedFlow.getFlowPathIds());
```
Comment (niksv) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java]: I'm not sure about pathIds. I guess we must get pathIds for reroute from original flow.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java [hunk 2]
```diff
@@ -279,9 +288,43 @@ public UnidirectionalFlow updateFlow(FlowDto flow) throws FlowNotFoundException
...
+        return updateFlowResult.getUpdatedFlow();
+    }
+
+    @VisibleForTesting
+    UpdateFlowResult.UpdateFlowResultBuilder prepareFlowUpdateResult(FlowDto flowDto, Flow flow) {
+        boolean changedStrategy = flowDto.getPathComputationStrategy() != null
+                && !flowDto.getPathComputationStrategy().equals(flow.getPathComputationStrategy());
+        boolean changedMaxLatency = flowDto.getMaxLatency() != null
+                && !flowDto.getMaxLatency().equals(flow.getMaxLatency());
+        boolean strategyIsMaxLatency = flowDto.getPathComputationStrategy() != null
```
[the remark this answers is not in the document]
Comment (rtretyak) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java]: noooooooo

[the remark this answers is not in the document]
Comment (niksv) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java]: oh. my bad. your name is correct

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java (hunk 1 above)
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java]: This flow update does not affect the flow paths at this point.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java [hunk 3]
```diff
@@ -279,9 +288,43 @@ public UnidirectionalFlow updateFlow(FlowDto flow) throws FlowNotFoundException
...
+                && flowDto.getPathComputationStrategy().equals(PathComputationStrategy.MAX_LATENCY)
+                || flowDto.getPathComputationStrategy() == null && flow.getPathComputationStrategy() != null
+                && flow.getPathComputationStrategy().equals(PathComputationStrategy.MAX_LATENCY);
```
[the remark this answers is not in the document]
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java]: Fixed.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java [hunk 4]
```diff
@@ -279,9 +288,42 @@ public UnidirectionalFlow updateFlow(FlowDto flow) throws FlowNotFoundException
 
             flowRepository.createOrUpdate(currentFlow);
 
-            return Optional.of(forwardFlow);
+            return Optional.of(result.updatedFlow(forwardFlow).build());
```
Comment (niksv) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java]: I think you need to add saving of `getPathComputationStrategy()` into `currentFlow`. like in lines 20 and 283

Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsService.java]: Saving of `getPathComputationStrategy()` is added here: #3036

**Pull request merged**